testing/src/cypress/e2e/02-institution/03-institution-profile.cy.ts

```diff
@@ -2,71 +2,165 @@ import DashboardInstitutionObject from "../../page-objects/Institution-objects/D
-    institutionObject.establishedDate().should("be.disabled");
-    institutionObject.firstNameInstitutionInputText().should("be.disabled");
-    institutionObject.lastNameInstitutionInputText().should("be.disabled");
-    institutionObject.emailInstitutionInputText().should("be.disabled");
-    institutionObject.phoneNumberInstitutionInputText().should("be.disabled");
-
     institutionObject
       .addressInstitutionInputText()
       .clear()
       .type(profileData.addressInput);
     institutionObject.cityInputText().clear().type(profileData.cityInput);
     institutionObject.countryInputText().click();
-    institutionObject
-      .countrySearchInputText(profileData.countryInput)
-      .type("{enter}");
+    institutionObject.countrySearchInputText(profileData.countryInput);
+
     institutionObject.provinceInputText().click({ force: true });
     institutionObject
       .provinceSearchInputText(profileData.provinceState)
       .type("{enter}");
     institutionObject.postalInputText().clear().type(profileData.postalInput);
   });
+
+  it("Verify that address line 1 should be a mandatory field", () => {
+    institutionObject.verifyThatFieldShouldNotBeEmpty(
+      institutionObject.addressInstitutionInputText(),
+      "Address line 1"
+    );
+  });
+
+  it("Verify that address line 1 should not be more than 100 chars", () => {
+    institutionObject.verifyThatFieldDoesNotAcceptMoreThanSpecificChars(
+      institutionObject.cityInputText(),
+      "City"
+    );
+  });
+
+  it("Verify that city should be a mandatory field", () => {
+    institutionObject.verifyThatFieldShouldNotBeEmpty(
+      institutionObject.cityInputText(),
+      "City"
+    );
+  });
+
+  it("Verify that city should not be more than 100 chars", () => {
+    institutionObject.verifyThatFieldDoesNotAcceptMoreThanSpecificChars(
+      institutionObject.addressInstitutionInputText(),
+      "City"
+    );
+  });
+
+  it("Verify that country field is a drop down and contains two values", () => {
+    const countryOptions = ["Canada", "other"];
+    institutionObject.countryInputText().click();
+    countryOptions.forEach((country) => {
+      institutionObject.verifyTheDropDownOptionsAreVisible(
+        institutionObject.countryDropdownMenu(),
+        country
+      );
+    });
+  });
+
+  it("Verify that selecting Canada from country dropdown makes the pre-populate the provinces available", () => {
+    institutionObject.countryInputText().click({ force: true });
+    institutionObject.countrySearchInputText(profileData.countryInput);
+    institutionObject.provinceInputText().click();
+    const provinceIds = Object.values(Provinces);
+    for (let i = 0; i < provinceIds.length; i++) {
+      institutionObject.provinceSearchInputText(Object.keys(Provinces)[i]);
+      institutionObject.verifyTheDropDownOptionsAreVisible(
+        institutionObject.provinceDropdownMenu(),
+        provinceIds[i]
```

Comment:
just a suggestion. we can use `.map() or .forEach()`, we can avoid `provinceIds[i]`.